### PR TITLE
Add backwards-compatible method for retrieving the multi-currency instance

### PR DESCRIPTION
**Pull request by ricardo**
Fixes 209-gh-Automattic/woopayments

#### Changes proposed in this Pull Request

During the MCCY back-end decoupling, we decided to move the singleton logic out of `MultiCurrency` (pcrvl0-1jD-p2#comment-668).

During our tests, we figured some extensions are calling the MCCY instance the wrong way – See comment from related issue. So, a backwards-compatible method is a requirement.

We also learned about a QIT warning that this PR addresses the same way as in [`register_script_with_dependencies`](https://github.com/Automattic/woocommerce-payments/blob/af3cc60d6e5dc41c20e384ce49e1cd05d7db89fa/includes/class-wc-payments.php#L1174).

#### Testing instructions

**WooCommerce Deposits fix:**
- Follow testing instructions from 209-gh-Automattic/woopayments and ensure the fatal error is no longer reproducible.

**Verify new QIT results:**
- p1726595299217699/1726481053.328369-slack-CGGCLBN58

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9455` or branch name `fix/backwards-compat-mccy-instance` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/backwards-compat-mccy-instance) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8ae02af1e16edee8a34242b012b8b7127db3fbf7
- Build time: 2024-09-27 20:12:52 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: multi-currency/src/MultiCurrency.php
Comment (lovo-h) [inline multi-currency/src/MultiCurrency.php:218]: Nit: We discussed this through Slack, but to reiterate, we may need to consider adding a deprecation message. We agreed that we need to discuss how to proceed further, so for now, we can leave it as-is.

**Review by lovo-h — APPROVED** [review]
LGTM 👍 

**Testing Instructions**
* ✅ `npm run test:php`, `npm run test:js`, and `npm run test:qit-security`.
* ✅ Reproduced original issue as described in 209-gh-Automattic/woopayments.
* ✅ Confirmed issue no longer exists in this branch (e.g. `fix/backwards-compat-mccy-instance`).
* ✅ Successfully checked out with the WooCommerce Deposits plugin.

**Comment by ricardo** [comment]
Thanks @lovo-h 🙏. I've added the `status: blocked` label to not merge this yet until we figure out if the MCCY branch shall be reverted for this release.

**Comment by ricardo** [comment]
I'm rebasing and merging this to `multi-currency-v2`, but the tests are failing due to `multi-currency-v2` not being synced with https://github.com/Automattic/woocommerce-payments/pull/9477 right now.

@lovo-h Could you take a look at syncing that PR with `multi-currency-v2` later? We might need to move the new methods from #9477 into the MCCY Utils.

**Pull request merged**
